platforms/core-configuration/model-core/src/main/java/org/gradle/internal/snapshot/impl/IsolatedArrayOfPrimitive.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.snapshot.impl;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.internal.hash.Hashable;
+import org.gradle.internal.hash.Hasher;
+import org.gradle.internal.isolation.Isolatable;
+import org.gradle.internal.snapshot.ValueSnapshot;
+import org.gradle.internal.snapshot.ValueSnapshottingException;
+
+import javax.annotation.Nullable;
+
+import static org.gradle.internal.snapshot.impl.AbstractValueProcessor.javaSerialized;
+import static org.gradle.internal.snapshot.impl.JavaSerializedValueSnapshot.javaDeserialized;
+
+@NonNullApi
+public class IsolatedArrayOfPrimitive implements Isolatable<Object>, Hashable {
```

Comment:
:grey_question: why does the class implement the `Hashable` directly? `Isolatable` already extends it.

Reply:
Riiight. Removed in 13457b8b963b2315ccbbbebb7f66f480f0166017, thanks!